x/staking/keeper/msg_server.go

```diff
@@ -35,8 +36,13 @@ var _ types.MsgServer = msgServer{}
 func (k msgServer) CreateValidator(goCtx context.Context, msg *types.MsgCreateValidator) (*types.MsgCreateValidatorResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
```

Comment:
Can we move this unwrapping after all the checks? This way we do not need to do it when the msg isn't valid.